### PR TITLE
OcTextInput: add defaultValue prop and make clear button null the value

### PR DESCRIPTION
**Pull request by dschmidt**
## Description
1) Fixed the clear button being visible even when element is disabled
2) Made clear button emit `null` instead of empty string
3) Made input/change event handling a bit more consistent
4) Added `defaultValue` prop that is for now passed as `placeholder` to the input field. 

Needs UX concept by @kulmann and @tbsbdr

## Related Issue

- Fixes one :x: from #1350 (`defaultValue`) and makes it more consistent with other components that also emit `null` on clear (which is not part of that table)

## Motivation and Context
Same as #1634 and #1635 

Instead of using native `placeholder` prop on the `input` element, I would propose to introduce the `defaultValue` property which for now uses `placeholder` but keeps that as implementation detail. That way I can already use the functionality in ownBrander 2.0 and will automatically benefit from a better implementation at some point in the future.

## Screenshots (if appropriate):
Only visual change: it's now possible to have an empty input field with clear button.
That is useful when a user overrides the default value with an empty string.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
     we don't emit empty strings anymore on clear, but afaict `clearButtonEnabled` is used nowhere, so this shouldn't break anything in ownCloud
- [ ] Technical debt
- [x] Tests

## Checklist:

- [x] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation added/updated

## Open tasks:

- [ ] ...

### CODE REVIEW

**Review by pascalwengerter — COMMENTED** [review]
LGTM and thanks for taking care, would like to get a final approval from @kulmann before merging

**Review by kulmann — APPROVED** [review]
LGTM 👍

**Pull request merged**
